src/snmalloc/mem/localalloc.h

```diff
@@ -1,7 +1,9 @@
 #pragma once
 
-#ifdef _MSC_VER
-#  define ALLOCATOR __declspec(allocator)
+#if defined(_MSC_VER)
+#  define ALLOCATOR __declspec(allocator) __declspec(restrict)
+#elif defined(__GNUC__)
```

Comment:
How about
```suggestion
#elif __has_attribute(malloc)
```
since I think both clang and gcc support that (though if CI says that we're supporting versions too old, well, so be it, I guess).

Reply:
well it seems only clang and gcc support this anyway https://www.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/compiler-reference/attributes.html seems i changes nothing in this case but I can always try.